apps/server/src/infra/sync/service/sync.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@nestjs/common';
+import { TspSyncStrategy } from '../tsp/tsp-sync.strategy';
+import { SyncStrategy } from '../strategy/sync-strategy';
+import { SyncStrategyTarget } from '../sync-strategy.types';
+
+@Injectable()
+export class SyncService {
+	strategies: Map<SyncStrategyTarget, SyncStrategy> = new Map<SyncStrategyTarget, SyncStrategy>();
+
+	constructor(private readonly tspSyncStrategy: TspSyncStrategy) {
+		this.registerStrategy(tspSyncStrategy);
+	}
+
+	protected registerStrategy(strategy: SyncStrategy) {
+		this.strategies.set(strategy.getType(), strategy);
+	}
+
+	public async startSync(target: string): Promise<void> {
+		const targetStrategy = target as SyncStrategyTarget;
```

Comment:
Might be better to us string as the key type of the map and then skip this cast.